content/ember-4-0-released.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Ember 4.0 released
+authors:
+  - jen-weber
+  - matthew-beale
+date: 2021-12-06T00:00:00.000Z
+tags:
+  - releases
+  - '2021'
+  - version-4-x
+---
+
+Today the Ember project is releasing version 4.0 of Ember.js, Ember Data, and
+Ember CLI. Ember's 4.0 release focuses
+the framework by removing long-deprecated APIs and support for legacy platforms.
+This blog post will help you understand how to upgrade and navigate any breaking
+changes.
+
+Although [Ember "Octane"](https://emberjs.com/editions/octane/) APIs have been
+the default for new applications since Ember 3.15, the framework has continued
+to support "Classic" framework features in accordance with our semantic
+versioning commitment. Ember 4.0 takes a step forward and drops
+already-deprecated classic APIs, however the foundational `EmberComponent` and
+`EmberObject`/`computed` APIs are not removed in this release.
+
+Additionally, today we're promoting Ember 3.28 to become Ember's latest
+Long-Term Support (LTS) release. Ember's LTS release process provides bugfix
+support for about 36 weeks, and security patches for about 54 weeks. See the
+[3.28 release post](https://blog.emberjs.com/ember-3-28-released) and [LTS
+documentation](https://emberjs.com/releases/lts/) for more details.
+
+For applications using LTS releases, upgrading directly to 4.0 is not
+encouraged. Best practice for those applications is to upgrade to Ember 3.28 LTS
+as soon as possible, then wait for the first LTS of the 4.0 series to be
+released. Ember 4.4 will be the first candidate.
+
+## Major Releases in Ember
+
+Starting with Ember 2.0, major versions of Ember have been about removal
+of deprecated APIs, and not about the introduction of new features or development styles.
+Ember 4.0 follows that tradition, and will contain no new features.
+
+Our plans for Ember 4.0 were originally announced in July 2021 in
+[The Road to Ember 4.0](https://blog.emberjs.com/the-road-to-ember-4-0/).
+
+## Trying Ember for the first time
+
+If you're interested in trying Ember for the
+first time today, get started by running:
+
+```bash
+# Make sure you are on the latest npm version, 8.x
+npm --version
+npx ember-cli@latest new my-project
+cd my-project
+npx ember serve
+# Then visit http://localhost:4200
```

Comment:
Please confirm these exact commands work as stated before merge (or someone comment here that they have done so).

Reply:
In this example, `npx ember serve` will work, because `ember` is the name of the executable within `/node_modules/.bin`. It's the first time I've seen it like this in any documentation though. Running e.g. `npx ember new` somewhere else will not work as it must be `npx ember-cli new` instead. After creating a new app, Ember CLI suggests to run `yarn start` or `npm start`. So that could be suggested as well instead of `npx ember serve`? And the docs use `ember server`.

Reply:
Yeah I really don't like suggesting `npm ember s` here when it is only using the local path. `npx`, to me, always implies running some thing from the package `ember`. In this case it is running something from "locally installed ember-cli" which I've never seen.

I don't know why we're introducing this new style of usage in this blog post. I'd like top revert it, again. @jenweber I think you brought it back after an earlier reversion, or had comments elsewhere here.

Reply:
Prior discussion:

* https://github.com/ember-learn/ember-blog/pull/1036#discussion_r753834945
* https://github.com/ember-learn/ember-blog/pull/1036#issuecomment-990034431

I think we agree not using `npx ember s` is the norm. I'm not clear on why @jenweber said `ember s` wouldn't work without the global installed. I thought it would be found from `node_modules/.bin` as well. If it isn't, then what we really want is `yarn run ember s`, etc.

Actually, I think `npm start` is the right thing to do here. I'm going to update to that.